feat(curriculum): Extend solution explanation for Convert HTML Entities challenge

guayom opened the pull request:
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

The [explanation](https://guide.freecodecamp.org/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/convert-html-entities/) for the advanced solution for [this challenge](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/convert-html-entities) is a little cumbersome.
It tried to explain the used method, but also tried to explain an alternative solution, both mixed together.

I decided to try to fix it the best I can. Here's my proposal:

To fix that, I moved everything related to the alternative solution to another paragraph.
Additionally, I added a more detailed explanation about this alternative solution.
Hopefully, with this fixes, both solutions will be clearer to the user.

I made sure that all the tests would pass if you use that solution.

guayom committed:
Extend solution explanation

The explanation for the advanced code solution was a little cumbersome.
It tried to explain the used method, but also tried to explain an alternative solution, both mixed together.
To fix that, I moved everything related to the alternative solution to another paragraph.
Additionally, I added a more detailed explanation about this alternative solution.
Hopefully, with this fixes, both solutions will be clearer to the user.

RandellDawson commented:
@guayom I like your explanation for the advanced solution.  As far as "alternative", I have a better idea.  Why don't you replace the existing Intermediate solution with your new solution and provide an explanation for your code.  Your's is much more concise than the current Intermediate Solution.

guayom commented:
Alright, I can do that!

guayom committed:
Replace intermediate solution

Replace the intermediate solution with a more concise one.

I took the opportunity to eliminate the indenting in all 3 snippets to remove unnecessary blank space on the left.
I also changed the wording on one of the instructions and fixed a typo.

thecodingaviator commented on guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/convert-html-entities/index.md:
```diff
@@ -35,39 +35,39 @@ title: Convert HTML Entities
+
+  var temp = str.split('');
+
+  // Since we are only checking for a few HTML elements I used a switch
```
```suggestion
  // Since we are only checking for a few HTML elements, use a switch
```

thecodingaviator commented on guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/convert-html-entities/index.md:
````diff
@@ -86,54 +86,63 @@ title: Convert HTML Entities
+```
 
-![:rocket:](https://forum.freecodecamp.com/images/emoji/emoji_one/rocket.png?v=3 ":rocket:") <a href='https://repl.it/CLnQ/0' target='_blank' rel='nofollow'>Run Code</a>
+![:rocket:](https://forum.freecodecamp.com/images/emoji/emoji_one/rocket.png?v=3 ":rocket:") <a href='https://repl.it/repls/WoozyDelightfulSoftwareengineer' target='_blank' rel='nofollow'>Run Code</a>
````
```suggestion
```
We're removing external links from now on, so this line should be deleted

thecodingaviator commented on guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/convert-html-entities/index.md:
````diff
@@ -86,54 +86,63 @@ title: Convert HTML Entities
+// test here
+convertHTML("Dolce & Gabbana");
 ```
 ![:rocket:](https://forum.freecodecamp.com/images/emoji/emoji_one/rocket.png?v=3 ":rocket:") <a href='https://repl.it/CLnR/0' target='_blank' rel='nofollow'>Run Code</a>
````
```suggestion
```
Same for this

guayom committed:
Remove external links

guayom committed:
Simplity regex declaration

I replaced `new RegExp()` in order to make the solution cleaner and shorter

RandellDawson reviewed (approved):
The new changes look good to me.
 @thecodingaviator If you like the changes, go ahead an merge.

thecodingaviator committed:
Update index.md

thecodingaviator reviewed (approved):
LGTM

thecodingaviator commented:
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝
